src/baseline.rs

```diff
@@ -7,22 +7,210 @@ use crate::dump::RustDocCommand;
+        unimplemented!()
+    }
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Clone)]
+enum CrateSource<'a> {
+    Registry { crate_: &'a crates_index::Version },
+    ManifestPath { path: &'a Path, name: String },
+}
+
+/// To get the rustdoc of the baseline, we first create a placeholder project somewhere
+/// with the baseline as a dependency, and run `cargo rustdoc` on it.
+fn create_placeholder_rustdoc_manifest(
+    crate_source: &CrateSource,
+) -> anyhow::Result<cargo_toml::Manifest<()>> {
+    use cargo_toml::*;
+
+    Ok(Manifest::<()> {
+        package: {
+            let mut package = Package::new("rustdoc", "0.0.0");
+            package.publish = Inheritable::Set(Publish::Flag(false));
+            Some(package)
+        },
+        workspace: Some(Workspace::<()>::default()),
+        lib: {
+            let product = Product {
+                path: Some("lib.rs".to_string()),
+                ..Product::default()
+            };
+            Some(product)
+        },
+        dependencies: {
+            let project_with_features: DependencyDetail = match crate_source {
+                CrateSource::Registry { crate_ } => DependencyDetail {
+                    // We need the *exact* version as a dependency, or else cargo will
+                    // give us the latest semver-compatible version which is not we want.
+                    // Fixes: https://github.com/obi1kenobi/cargo-semver-checks/issues/261
+                    version: Some(format!("={}", crate_.version())),
+                    features: crate_features::get_all_crate_features_from_registry(crate_),
+                    ..DependencyDetail::default()
+                },
+                CrateSource::ManifestPath { path, .. } => DependencyDetail {
+                    path: Some(
+                        // The manifest will be saved in some other directory,
+                        // so for convenience, we're using absolute paths.
+                        path.canonicalize()
+                            .with_context(|| "failed to canonicalize manifest path")?
+                            .to_str()
+                            .with_context(|| "manifest path is not valid UTF-8")?
+                            .to_string(),
+                    ),
+                    features: crate_features::get_all_crate_features_from_manifest(path),
+                    ..DependencyDetail::default()
+                },
+            };
+            let mut deps = DepsSet::new();
+            deps.insert(
+                match crate_source {
+                    CrateSource::Registry { crate_ } => crate_.name().to_string(),
+                    CrateSource::ManifestPath { name, path: _ } => name.to_string(),
+                },
+                Dependency::Detailed(project_with_features),
+            );
+            deps
+        },
+        ..Default::default()
+    })
+}
+
+fn save_placeholder_rustdoc_manifest(
```

Comment:
This could be an extension trait on `cargo_toml::Manifest`:

```rust
trait ManifestExt {
    fn save_to(&self, path: &Path) -> io::Result<>;
}

impl ManifestExt for cargo_toml::Manifest { }
```

Just a different way of doing things though :)

Reply:
Extension traits always felt a bit too magical for my taste, except in cases like `itertools` where the extension methods are truly general-purpose and applicable everywhere. I appreciate you highlighting the option, though!

Reply:
@obi1kenobi do I understand correctly that you propose to *not* implement this change? I'm confused.

Reply:
I think implementing this change would make the code a bit harder to understand -- this code removes a `Cargo.lock` in the same directory, which is a fix specific only to the current use case, but that change would suggest that it is a function for general use.

Reply:
Yes, I wouldn't do it.